consumer/consumer.go

```diff
@@ -799,10 +799,10 @@ func (dc *defaultConsumer) processPullResult(mq *primitive.MessageQueue, result
 
 		// filter message according to tags
 		msgListFilterAgain := msgs
-		if len(data.Tags) > 0 && data.ClassFilterMode {
+		if data.Tags.Len() > 0 && data.ClassFilterMode {
```

Comment:
Why make msgListFilterAgain a length of len(msgs).
Maybe it should be initialized to zero.

Reply:
done @ShannonDing